api/v1alpha4/common_types.go

```diff
@@ -52,6 +52,11 @@ const (
 	// on the reconciled object.
 	PausedAnnotation = "cluster.x-k8s.io/paused"
 
+	// DisableMachineCreate is an annotation that can be used to signal a MachineSet to stop creating new machines.
+	// It is utilized in the OnDelete MachineDeploymentStrategy to allow the MachineDeployment controller to scale down
+	// older MachineSets when Machines are deleted and add the new replicas to the latest MachineSet.
+	DisableMachineCreate = "cluster.x-k8s.io/disable-machine-create"
```

Comment:
Could we pause the MachineSet instead of introducing a new annotation?

Reply:
I guess rewording a bit: is there any way for the MachineSet to understand this state rather than introducing a new annotation altogether?

Reply:
I put a summary of why that will not work in this comment:
https://github.com/kubernetes-sigs/cluster-api/pull/4346#issuecomment-811088944

If we adjusted how the "paused" annotation worked though we could do that but I viewed that as a breaking change

Reply:
We might want to open an issue regarding pause and the update of statuses like @detiber suggested

Reply:
issue opened: https://github.com/kubernetes-sigs/cluster-api/issues/4473